sql/src/main/java/io/crate/analyze/symbol/Symbol.java

```diff
@@ -42,4 +43,12 @@ public static boolean isLiteral(Symbol symbol, DataType expectedType) {
     public String toString() {
         return representation();
     }
+
+    /**
+     * Typically, we only allow casting of Literals and undefined types.
+     */
+    @Override
+    public boolean canBeCasted() {
+        return valueType() != null && valueType().id() == UndefinedType.INSTANCE.id();
```

Comment:
imho valueType shouldn't be nullable. So far we've never any null checks if we use valueType